src/librustdoc/test.rs

```diff
@@ -481,7 +481,7 @@ pub fn make_test(s: &str,
     if dont_insert_main || already_has_main {
         prog.push_str(everything_else);
     } else {
-        prog.push_str("fn main() {\n");
+        prog.push_str("fn main() -> impl std::process::Termination {\n");
```

Comment:
We should probably also check for `#![no_std]` being in the doctest, and switch to `core::process` as appropriate.

Reply:
If the trait is in core you can just use `core`, I *think* we want to make both of them available? cc @petrochenkov @joshtriplett @aturon

Reply:
:+1:

Reply:
It's now using `core::process::Termination`.

Reply:
Which unfortunately isn't there yet.